### PR TITLE
[bugfix] More predictable month diffs at end-of-month

### PR DESCRIPTION
**Pull request by asenousy**
this resolves both #3980 & #3029 covering all end of month diff edge cases
here is [codepen](https://codepen.io/asenousy/pen/NBgoOz) for bugs fixed

this is my first ever pull request :) so please let me know if I need to adjust anything, thanks

### CODE REVIEW

File: src/lib/moment/diff.js [hunk 1]
```diff
@@ -36,19 +40,45 @@ export function diff (input, units, asFloat) {
...
+    }
+    var dur = createDuration(val, 'months');
+    var months = absRound(dur._months);
+    var dayInMonth = a.date();
+    var newMonth = get(a, 'Month') + months;
+    var maxDays = daysInMonth(a.year(), newMonth);
+    // if End of Month, should ignore time difference
+    if (dayInMonth > maxDays || isEndOfMonth(a)) {
+        dayInMonth = maxDays;
+        var h = b._d['get' + (a._isUTC ? 'UTC' : '') + 'Hours']();
```
Comment (kakasahebKK) [inline src/lib/moment/diff.js]: Why not to store the condition result in a const and reuse it ? such as:
```suggestion
        const utcChecked = a._isUTC ? 'UTC' : '';
        const getUtcChecked = 'get' + utcChecked;
        const setUtcChecked = 'set' + utcChecked;
        const h = b._d[getUtcChecked + 'Hours']();
```
and so on ..

Comment (asenousy) [reply src/lib/moment/diff.js]: done, thanks :)

Comment (asenousy) [reply src/lib/moment/diff.js]: sorry forgot to resolve it :)

**Comment by antonina-cherednichenko** [comment]
are there any plans to merge this? Thank you

**Comment by asenousy** [comment]
That would be great, @kakasahebKK @ichernev @marwahaha please let me know if there is anything more I can do to get this merged :) thanks

**Comment by ichernev** [comment]
Merged in 813eb328bec31297a58d4f315bc80e9b4bea327a

**Comment by ichernev** [comment]
I mostly took the tests and used different, more understandable logic to achieve the same result.

**Pull request closed without merging**